wp/lib/bap_wp/src/constraint.ml

```diff
@@ -323,7 +377,39 @@ let get_refuted_goals ?filter_out:(filter_out = []) (constr : t)
             Seq.append (worker c current_path current_registers olds news) accum)
     | Subst (e, o, n) ->
       let n' = List.map n ~f:(fun x -> Expr.substitute x olds news |> eval_model_exn model) in
-      worker e current_path current_registers (olds @ o) (news @ n')
+      if (List.length n') = 1 (* It appears we generate overwhelmingly size 1 lists *)
```

Comment:
Looks like there's quite a bit of code duplication here. Is this any different than what's above? Can you re-factor this?

It's gonna be a nightmare to debug if we have to maintain them in parallel.